batch/batch/batch.py

```diff
@@ -188,24 +191,29 @@ class Job:
 
         pod_template = kube.client.V1Pod(
             metadata=kube.client.V1ObjectMeta(
-                generate_name='batch-{}-job-{}-{}-'.format(self.batch_id, self.job_id, self._current_task.name),
+                name=self._pod_name,
                 labels={'app': 'batch-job',
                         'hail.is/batch-instance': INSTANCE_ID,
+                        'batch_id': str(self.batch_id),
+                        'job_id': str(self.job_id),
+                        'task': self._current_task.name,
+                        'user': self.user,
                         'uuid': uuid.uuid4().hex
                         }),
             spec=pod_spec)
 
         pod, err = await app['k8s'].create_pod(body=pod_template)
         if err is not None:
+            if err.status == 409:
+                log.info(f'pod already exists for job {self.id}')
+                await db.jobs.update_record(*self.id, state='Running')
```

Comment:
this could race with a pod completing and change a state from success to running, right?

Reply:
Working on a solution with #6523